Release v1.5.0 (#4646)
operator-framework/operator-sdk: website/content/en/docs/upgrading-sdk-version/v1.5.0.md

```console $ cat PROJECT version: 3-alpha resources: - crdVersion: v1 ... $ operator-sdk alpha config-3alpha-to-3 Your PROJECT config file has been converted from version 3-alpha to 3. Please make sure all config data is correct. $ cat PROJECT version: "3" resources: - api:
    crdVersion: v1
... ```

_See [#4613](https://github.com/operator-framework/operator-sdk/pull/4613) for more details._

## (go/v3) Upgrade controller-runtime to v0.7.2

In your go.mod file, upgrade `sigs.k8s.io/controller-runtime` to v0.7.2.

_See [#4626](https://github.com/operator-framework/operator-sdk/pull/4626) for more details._

## (go/v3) Add a `system:controller-manager` ServiceAccount to your project.

A non-default ServiceAccount `controller-manager` is scaffolded on `operator-sdk init`, to improve security for operators installed in shared namespaces. To add this ServiceAccount to your project, do the following:

> ```sh # Create the ServiceAccount. cat <<EOF > config/rbac/service_account.yaml apiVersion: v1 kind: ServiceAccount metadata:
>   name: controller-manager
>   namespace: system
> EOF
> # Add it to the list of RBAC resources. echo "- service_account.yaml" >> config/rbac/kustomization.yaml
> # Update all RoleBinding and ClusterRoleBinding subjects that reference the operator's ServiceAccount. find config/rbac -name *_binding.yaml -exec sed -i -E 's/  name: default/  name: controller-manager/g' {} \;
> # Add the ServiceAccount name to the manager Deployment's spec.template.spec.serviceAccountName. sed -i -E 's/([ ]+)(terminationGracePeriodSeconds:)/\1serviceAccountName: controller-manager\n\1\2/g' config/manager/manager.yaml ```

Reviewer: ````suggestion
```sh
# Create the ServiceAccount.
cat <<EOF > config/rbac/service_account.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: controller-manager
  namespace: system
EOF

# Add it to the list of RBAC resources.
echo "- service_account.yaml" >> config/rbac/kustomization.yaml

# Update all RoleBinding and ClusterRoleBinding subjects that reference the operator's ServiceAccount.
find config/rbac -name *_binding.yaml -exec sed -i -E 's/  name: default/  name: controller-manager/g' {} \;

# Add the ServiceAccount name to the manager Deployment's spec.template.spec.serviceAccountName.
sed -i -E 's/([ ]+)(terminationGracePeriodSeconds:)/\1serviceAccountName: controller-manager\n\1\2/g' config/manager/manager.yaml
```
````

The changes should look like:

> ```diff # config/manager/manager.yaml

Reviewer: ````suggestion
```diff
# config/manager/manager.yaml
````

           requests:
             cpu: 100m
             memory: 20Mi
+      serviceAccountName: controller-manager
       terminationGracePeriodSeconds: 10

# config/rbac/auth_proxy_role_binding.yaml
   name: proxy-role
 subjects:
 - kind: ServiceAccount
-  name: default +  name: controller-manager
   namespace: system

# config/rbac/kustomization.yaml
 resources:
+- service_account.yaml
 - role.yaml
 - role_binding.yaml
 - leader_election_role.yaml

# config/rbac/leader_election_role_binding.yaml
   name: leader-election-role
 subjects:
 - kind: ServiceAccount
-  name: default +  name: controller-manager
   namespace: system

# config/rbac/role_binding.yaml
   name: manager-role
 subjects:
 - kind: ServiceAccount
-  name: default +  name: controller-manager
   namespace: system

> # config/rbac/service_account.yaml +apiVersion: v1 +kind: ServiceAccount +metadata: +  name: controller-manager +  namespace: system ```

Reviewer: ````suggestion
# config/rbac/service_account.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: controller-manager
+  namespace: system
```
````

_See [#4626](https://github.com/operator-framework/operator-sdk/pull/4626) for more details._

> ## For Ansible/Helm-based operators, swap the paths of the probes in config/manager/manager.yaml.

Reviewer: ```suggestion
## (ansible/v1, helm/v1) Swap the paths of liveness/readiness probes in `config/manager/manager.yaml`.
```

The liveness and readiness probe endpoints were incorrectly named, although this mismatch will not affect their behavior. To fix, swap the `readinessProbe` and `livenessProbe` HTTP paths in `config/manager/manager.yaml`:

> ```

Reviewer: ````suggestion

```yaml
````

```sh
# Create the ServiceAccount.
cat <<EOF > config/rbac/service_account.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: controller-manager
  namespace: system
EOF

# Add it to the list of RBAC resources.
echo "- service_account.yaml" >> config/rbac/kustomization.yaml

# Update all RoleBinding and ClusterRoleBinding subjects that reference the operator's ServiceAccount.
find config/rbac -name *_binding.yaml -exec sed -i -E 's/  name: default/  name: controller-manager/g' {} \;

# Add the ServiceAccount name to the manager Deployment's spec.template.spec.serviceAccountName.
sed -i -E 's/([ ]+)(terminationGracePeriodSeconds:)/\1serviceAccountName: controller-manager\n\1\2/g' config/manager/manager.yaml
```

The changes should look like:
```diff 
# config/manager/manager.yaml
           requests:
             cpu: 100m
             memory: 20Mi
+      serviceAccountName: controller-manager
       terminationGracePeriodSeconds: 10

# config/rbac/auth_proxy_role_binding.yaml
   name: proxy-role
 subjects:
 - kind: ServiceAccount

> -  name: default +  name: controller-manager

Reviewer: ```suggestion
-  name: default
+  name: controller-manager
```

   namespace: system

# config/rbac/kustomization.yaml
 resources:
+- service_account.yaml
 - role.yaml
 - role_binding.yaml
 - leader_election_role.yaml

# config/rbac/leader_election_role_binding.yaml
   name: leader-election-role
 subjects:
 - kind: ServiceAccount

> -  name: default +  name: controller-manager

Reviewer: ```suggestion
-  name: default
+  name: controller-manager
```

   namespace: system

# config/rbac/role_binding.yaml
   name: manager-role
 subjects:
 - kind: ServiceAccount

> -  name: default +  name: controller-manager

Reviewer: ```suggestion
-  name: default
+  name: controller-manager
```
